### DISCUSSION TITLE
Access MarkdocNextJsPageProps in `article`

### DISCUSSION THREAD

**Discussion by ReeceHumphreys** [markdoc/markdoc#279] [Q&A] [answered]
I would like to embed a table of contents inside of the article element that markdoc renders. I have a document component and a custom defined document type:

```tsx
export function Document({ children }) {
  return (
    <div className="document">
      <article>
        {children}
        <style jsx>
          {`
            article {
              background: white;
            }
          `}
        </style>
      </article>
    </div>
  );
}

```

```ts
import { Tag, nodes } from '@markdoc/markdoc';
import { Document } from '../../components/Document';

export default {
  ...nodes.document,
  render: Document,
  transform(node, config) {
    return new Tag(
      this.render,
      { config: config },
      node.transformChildren(config)
    );
  }
};
```

It seems that the most straight forward way to generate the TOC would be like the nextjs starter template where you pull the current page markdoc content from pageProps `collectHeadings(pageProps.markdoc.content)`.

However, It is unclear to me how I would get the `pageProps` from `_app.tsx` to the Document component. Do you guys have any suggestions or guidence?

**Comment by mfix-stripe** [comment, marked as answer]
`pageProps.markdoc.content` is very similar to what you would get from calling `node.transformChildren(config)`
